gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java

```diff
@@ -616,23 +689,75 @@ protected void addFiles(GobblinMetadataChangeEvent gmce, Map<String, Collection<
   /**
    * Method to get dataFiles with metrics information
    * This method is used to get files to be added to iceberg
+   * if completeness is enabled a new field (late) is added to table schema and partition spec
+   * computed based on datepartition and completion watermark
    * This method will call method {IcebergUtils.getIcebergDataFileWithMetric} to get DataFile for specific file path
    */
-  private Set<DataFile> getIcebergDataFilesToBeAdded(List<org.apache.gobblin.metadata.DataFile> files,
+  private Set<DataFile> getIcebergDataFilesToBeAdded(Table table, TableMetadata tableMetadata, GobblinMetadataChangeEvent gmce, List<org.apache.gobblin.metadata.DataFile> files,
       PartitionSpec partitionSpec, Map<String, Collection<HiveSpec>> newSpecsMap, Map<Integer, Integer> schemaIdMap) {
     Set<DataFile> dataFiles = new HashSet<>();
     for (org.apache.gobblin.metadata.DataFile file : files) {
       try {
-        StructLike partition = getIcebergPartitionVal(newSpecsMap.get(new Path(file.getFilePath()).getParent().toString()),
-            file.getFilePath(), partitionSpec);
-        dataFiles.add(IcebergUtils.getIcebergDataFileWithMetric(file, partitionSpec, partition, conf, schemaIdMap));
+        Collection<HiveSpec> hiveSpecs = newSpecsMap.get(new Path(file.getFilePath()).getParent().toString());
```

Comment:
For this part, I feel we have many assumptions, i.e. we only have one partition column, and it's datepartition. But it's not always hold true and GMIP was supposed to support all generic use cases. Do you think it's make sense to have another icebergWriterWithCompletenessCheck for this new feature?

Reply:
Agree with the assumptions here but this logic is only triggered when completeness is enabled. Creating a new writer extending this writer will need a bigger refactor which should probably be part of another PR.

Reply:
Yeah, just want to avoid confusing people when reading through the code. Can you create a follow up ticket to refactor the class?

Reply:
https://issues.apache.org/jira/browse/GOBBLIN-1540